src/pages/tBTC/Bridge/index.tsx

```diff
@@ -46,24 +34,20 @@ const TBTCBridge: PageComponent = (props) => {
     )
   }, [dispatch, account])
 
+  const shouldRenderSidebars = pathname.startsWith("/tBTC/mint/deposit")
+
   return (
-    <Grid
-      gridTemplateAreas={gridTemplateAreas}
-      gridTemplateColumns={{ base: "1fr", xl: "25% 1fr" }}
-      gap="5"
+    <PageLayout
+      renderTop={<h1>TODO: My balance component</h1>}
+      renderLeft={shouldRenderSidebars ? <MintingTimeline /> : null}
+      renderRight={
+        shouldRenderSidebars ? (
+          <h2>TODO: Transaction history + Knowledgebase component</h2>
+        ) : null
+      }
     >
-      <Box gridArea="main" minW="0">
-        <MintUnmintNav w="100%" gridArea="nav" mb="5" pages={props.pages!} />
-        <Outlet />
-      </Box>
-      <Box gridArea="aside">
-        <TbtcBalanceCard gridArea="balance-card" mb="5" />
-        <BridgeActivityCard
-          data={bridgeActivity}
-          isFetching={isBridgeActivityFetching}
-        />
-      </Box>
-    </Grid>
+      <Outlet />
+    </PageLayout>
```

Comment:
In this particular case it's pointless to leverage compound components pattern. 
The major advantage of this pattern is a **loose structure** with **communication between children**. We don't want loose structure in Layout, we know strictly what is the expected structure. Moreover we don't need communication. We just want to render specific content in specific area.

The example you provided does not present compound components pattern. Components don't have any shared state - they "don't know" about each other. They are just imported as an object so you can use dot notation while referring. Compound components share the same state (usually context) and are aware of state changes managed by the parent node (it's possible to make it bi-directional too).

[Render slots](https://daveceddia.com/pluggable-slots-in-react-components/) is the proper architectural pattern to use in this case.

> I strongly believe that passing components through props can get cumbersome

I don't understand. Render props are typed to accept only `ReactNode` exactly the same as children which in fact is just specific prop you define by passing value between opening and closing tags. 
```jsx
<Component>
  <div>content</div>
</Component>
```
is exactly the same as
```jsx
<Component children={<div>content</div>} />
```
It's impossible to mess it up, especially when it's strongly typed. I also took care of documentation and described each prop and component to prevent confusions when consuming component.

To sum up:
- compound components are suitable when you want loose structure and communication with shared state
- render slots are suitable when "children" is not enough - when you need to define multiple render areas to be present in specific places, you can treat is as "alternative children".

Here is [example of compound components pattern](https://github.com/kpyszkowski/Nardo-Primitives/blob/feature/tabs/src/components/Tabs/Tabs.tsx) I have created some time ago. It's not ideal buy presents key concepts of the pattern.

Reply:
I just feel like the approach I proposed is a little bit more idiomatic and more natural to React in general. But we can have it your way too, I don't mind. Both solutions are pretty similar to each other.